library/src/main/java/com/cleveroad/fanlayoutmanager/FanLayoutManager.java

```diff
@@ -874,7 +874,28 @@ public void straightenSelectedItem(Animator.AnimatorListener listener) {
         if (viewToRotate != null) {
 
             // start straight animation
-            animationHelper.straightenView(viewToRotate, listener);
+            animationHelper.straightenView(viewToRotate, listener, new
```

Comment:
You don't need to pass a new instance just for changing the `isSelectedItemStraightened` flag. Just wrap an existing `listener` with a new one. An example of better approach:

```JAVA
animationHelper.straightenView(viewToRotate, new AnimatorListenerAdapter() { 
	@Override
	public void onAnimationStart(Animator animation) {
		listener.onAnimationStart(animation);
	}

	@Override
	public void onAnimationEnd(Animator animation) {
		isSelectedItemStraightened = false;
		listener.onAnimationEnd(animation);
	}

	@Override
	public void onAnimationCancel(Animator animation) {
		listener.onAnimationCancel(animation);
	}

	@Override
	public void onAnimationRepeat(Animator animation) {
		listener.onAnimationRepeat(animation);
	}
});
```